Common/Data/Custom/Benzinga/BenzingaNews.cs

```diff
@@ -0,0 +1,226 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using Newtonsoft.Json;
+using QuantConnect.Util;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace QuantConnect.Data.Custom.Benzinga
+{
+    /// <summary>
+    /// News data powered by Benzinga - https://benzinga.com/
+    /// </summary>
+    [JsonObject(MemberSerialization.OptIn)]
+    public class BenzingaNews : IndexedBaseData
+    {
+        /// <summary>
+        /// Title of the article published
+        /// </summary>
+        [JsonProperty("title")]
+        public string Title { get; set; }
+
+        /// <summary>
+        /// Contents of the article
+        /// </summary>
+        [JsonProperty("description")]
+        public string Contents { get; set; }
+
+        /// <summary>
+        /// Date the article was published
+        /// </summary>
+        [JsonProperty("pubDate")]
+        public DateTime PublicationDate { get; set; }
+
+        /// <summary>
+        /// Author of the article
+        /// </summary>
+        [JsonProperty("dc:creator")]
+        public string Author { get; set; }
+
+        /// <summary>
+        /// Categories that the article belongs to
+        /// </summary>
+        [JsonProperty("category"), JsonConverter(typeof(SingleValueListConverter<BenzingaCategory>))]
+        public List<BenzingaCategory> Category { get; set; }
+
+        /// <summary>
+        /// Unique ID assigned to the article by Benzinga
+        /// </summary>
+        [JsonProperty("bz:id")]
+        public string Id { get; set; }
```

Comment:
Lets make it a habit to fill in the <example> xml too for all of these so we can review/document properly. Is this a string id?

Reply:
Going through all the fields (as a quick self review):

* `Title: string`
* `Contents: string`
* `PublicationDate: DateTime`
* `Author: string`
* `Category: List<BenzingaCategory>`
    - `Resource: string` - can be links with ticker included in them. Decided against using enum here
    - `Description: string` - same as Resource
* `Id: string` - Only numerical representations appear in the raw data. Will update to int
* `RevisionId: string` - Only numerical representations appear in the raw data. Will update to int
* `Symbols: List<BenzingaSymbolData>`
    - `Sentiment: decimal?` - Not all fields contain sentiment alongside with them. Maybe I should default to 0 if sentiment is missing? But at the same time, a sentiment of 0 means something if it's been deliberately included...
    - `Exchange: string` - Thought about making this an enum, but might be vendor specific since we have some exchanges like "ETF" in the raw data
    - `Symbol: Symbol`
* `RevisionDate: DateTime`
* `Metadata: BenzingaMetadata`
    - `IsPro: bool`
    - `FirstRun: bool`
    - `Kind: string` - Always appears as "source" in the raw data, but I believe having it as an enum value isn't right since we'd have an enum with only one value, and it will break if another value shows up during processing

Just a quick synopsis of the fields present and some additional notes containing some design considerations.